libraries/AP_AIS/AP_AIS.cpp

```diff
@@ -95,17 +95,44 @@ void AP_AIS::init()
     }
 
     _uart = AP::serialmanager().find_serial(AP_SerialManager::SerialProtocol_AIS, 0);
-    if (_uart == nullptr) {
-        return;
+    if (_uart != nullptr) {
+        _uart->begin(AP::serialmanager().find_baudrate(AP_SerialManager::SerialProtocol_AIS, 0));
     }
 
-    _uart->begin(AP::serialmanager().find_baudrate(AP_SerialManager::SerialProtocol_AIS, 0));
 }
 
 // update AIS, expected to be called at 20hz
 void AP_AIS::update()
 {
-    if (!_uart || !enabled()) {
+    if (!enabled()) {
+        return;
+    }
+
+    // remove expired items from the list
+    const uint32_t now_ms =  AP_HAL::millis();
+    const uint32_t timeout_ms = MAX(_time_out.get(), 1) * 1000;
+    for (uint16_t i = 0; i < _list.max_items(); i++) {
+        if (_list[i].last_update_ms != 0 && now_ms - _list[i].last_update_ms >= timeout_ms) {
+            clear_list_item(i);
+        }
+    }
+
+#if 1
+    if (_type.get() == 2) {
+        mavlink_ais_vessel_t packet {};
+
+        packet.MMSI = 131313131;
+        packet.lat = -35.3621640 *1e7;
+        packet.lon = 149.1658536 *1e7;
+        strcpy(packet.callsign, "test1");
+        strcpy(packet.name, "test2");
+
+        handle_message(packet);
+    }
+#endif
```

Comment:
Remove testing code (once testing is done ;) ).

Reply:
yup! Gone